src/datafusion/parser.rs

```diff
@@ -219,6 +223,20 @@ impl<'a> DFParser<'a> {
         })))
     }
 
+    pub fn parse_truncate(&mut self) -> Result<Statement, ParserError> {
+        let table_name: ObjectName = if self.parser.parse_keyword(Keyword::TABLE) {
+            self.parser.parse_object_name(true)?
+        } else {
+            return self.expected("TABLE as a TRUNCATE target", self.parser.peek_token());
+        };
```

Comment:
You can handle the error first here to simplify the flow and reduce the need for nested code.

```suggestion
    if !self.parser.parse_keyword(Keyword::TABLE) {
        return self.expected("TABLE as a TRUNCATE target", self.parser.peek_token());
    }

    let table_name = self.parser.parse_object_name(true)?;
```